internal/data/outboxeventsrepository.go

```diff
@@ -18,3 +20,38 @@ func PublishOutboxEvent(tx *gorm.DB, event *model_legacy.OutboxEvent) error {
 	}
 	return nil
 }
+
+// HandleWorkspaceRelationOutboxEvents generates both create and delete outbox events for workspace relation changes
+// This ensures new workspace relations are added before old ones are deleted, preventing access gaps
+// Following the add-first-then-delete approach: brief period of extra access is acceptable
+func HandleWorkspaceRelationOutboxEvents(tx *gorm.DB, resource model_legacy.Resource, oldWorkspaceId string, namespace string, txid string) error {
```

Comment:
Repository layer should not have any business logic, it should just handle simple reads and writes to the database. This code should not live here.